apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[slug]/(active)/components/client/auth-options-form.client.tsx

```diff
@@ -25,65 +38,212 @@ export function AuthOptionsForm({ ecosystem }: { ecosystem: Ecosystem }) {
+              "hover:cursor-pointer hover:text-foreground",
+            )}
+          >
+            <Checkbox
+              checked={ecosystem.authOptions?.includes(option)}
+              onClick={() => {
+                if (ecosystem.authOptions?.includes(option)) {
+                  setMessageToConfirm({
+                    title: `Are you sure you want to remove ${option.slice(0, 1).toUpperCase() + option.slice(1)} as an authentication option for this ecosystem?`,
+                    description:
+                      "Users will no longer be able to log into your ecosystem using this option. Any users that previously used this option will be unable to log in.",
+                    authOptions: ecosystem.authOptions?.filter(
+                      (o) => o !== option,
+                    ),
+                  });
+                } else {
+                  setMessageToConfirm({
+                    title: `Are you sure you want to add ${option.slice(0, 1).toUpperCase() + option.slice(1)} as an authentication option for this ecosystem?`,
+                    description:
+                      "Users will be able to log into your ecosystem using this option. If you later remove this option users that used it will no longer be able to log in.",
+                    authOptions: [...ecosystem.authOptions, option],
+                  });
+                }
+              }}
+            />
+            {option.slice(0, 1).toUpperCase() + option.slice(1)}
+          </CheckboxWithLabel>
+        ))}
+        <ConfirmationDialog
+          open={!!messageToConfirm}
+          onOpenChange={(open) => {
+            if (!open) {
+              setMessageToConfirm(undefined);
+            }
+          }}
+          title={messageToConfirm?.title}
+          description={messageToConfirm?.description}
+          onSubmit={() => {
+            invariant(
+              messageToConfirm,
+              "Must have message for modal to be open",
+            );
+            updateEcosystem({
+              ...ecosystem,
+              authOptions: messageToConfirm.authOptions,
+            });
+          }}
+        />
+      </div>
+      <CustomAuthOptionsForm ecosystem={ecosystem} />
+    </div>
+  );
+}
+
+function CustomAuthOptionsForm({ ecosystem }: { ecosystem: Ecosystem }) {
+  const form = useForm({
+    defaultValues: {
+      customAuthEndpoint: ecosystem.customAuthOptions?.authEndpoint?.url,
+      customHeaders: ecosystem.customAuthOptions?.authEndpoint?.headers,
+    },
+  });
+  const { fields, remove, append } = useFieldArray({
+    control: form.control,
+    name: "customHeaders",
+  });
+  const { mutateAsync: updateEcosystem, isPending } = useUpdateEcosystem({
+    onError: (error) => {
+      const message =
+        error instanceof Error ? error.message : "Failed to update ecosystem";
+      toast.error(message);
+    },
+    onSuccess: () => {
+      toast.success("Custom Auth Options updated");
+    },
```

Comment:
this can be improved to use `toast.promise` etc @MananTank or @kien-ngo  maybe you can clean it up down the line - for now this is fine